app/services/ai_backends/open_ai.rb

```diff
@@ -0,0 +1,51 @@
+class AiBackends::OpenAi
+  attr :ai_model
+  attr :client
+
+  # Rails system tests don't seem to allow mocking because the server and the
+  # test are in separate processes.
+  #
+  # In regular tests, mock this method or the TestClients::OpenAi class to do
+  # what you want instead.
+  def self.client
+    if Rails.env.test?
+      TestClients::OpenAi
+    else
+      OpenAI::Client
+    end
+  end
```

Comment:
@krschacht this isn't my favorite -- special casing the test environment feels like playing with fire to me. But this abstraction layer did clean up the interface nicely anyway so I'll call it even.